File: 221111-dojo/java/app/src/main/java/Solution.java
import java.util.*;

class Solution {
    public int[] solution(String msg) {
        int[] answer = {};

        // 사전 만들기 -> 해쉬맵으로 해보자.
        HashMap<Integer, String> dictionary = new HashMap<>();

        for (int i = 1; i <= 26; i += 1) {
            dictionary.put(i, (char) (24 + i) + "");
        }

        return answer;
    }
}

File: 221111-dojo/javascript/solution.js
export function initDictionary() {
  const dictionary = [];

  for (let i = 'A'.charCodeAt(0); i <= 'Z'.charCodeAt(0); i += 1) {
    dictionary.push(String.fromCharCode(i));
  }

  return dictionary;
}

export default function solution(message) {
  const answer = [];

  // 사전 준비
  const dictionary = initDictionary();

  let i = 0;

  while (i < message.length) {
    let j = 1;

    while (dictionary.indexOf(message.substring(i, i + j)) !== -1 && i + j <= message.length) {
      j += 1;
    }

    dictionary.push(message.substring(i, i + j));

    answer.push(dictionary.indexOf(message.substring(i, i + j - 1)) + 1);
    i += j - 1;
  }

  return answer;
}
